scripts/hls/comp_hls.tcl.in

```diff
@@ -48,56 +48,59 @@ if {$cfg(load_apps) eq 1} {
 
             set tmp_val "[format "\$vfpga_c$i\_$j"]" 
             set tmp_path "[format "%s" $tmp_val]"
-            set cmd "[format "set tmp_path $tmp_path"]"
+            set cmd "[format "set tmp_path \"$tmp_path\""]"
             eval $cmd
+            set src_dirs [split $tmp_path]
 
             # Compile all HLS cores
-            if {[file isdirectory "${CMAKE_SOURCE_DIR}/$tmp_path/hls"]} {
-                set krnls [glob -nocomplain -tails -directory "${CMAKE_SOURCE_DIR}/$tmp_path/hls" -type d *]
-
-                foreach krnl $krnls {
-                    # Copy all files
-                    file mkdir "$build_dir/$project\_config_$i/user\_c$i\_$j/hdl/ext/$krnl\_hls"
-                    set files_hdl [glob -nocomplain "${CMAKE_SOURCE_DIR}/$tmp_path/hls/$krnl/*.hpp" "${CMAKE_SOURCE_DIR}/$tmp_path/hls/$krnl/*.cpp" "${CMAKE_SOURCE_DIR}/$tmp_path/hls/$krnl/*.h" "${CMAKE_SOURCE_DIR}/$tmp_path/hls/$krnl/*.c"]
-                    if {[llength $files_hdl]} {
-                        file copy -force {*}$files_hdl "$build_dir/$project\_config_$i/user\_c$i\_$j/hdl/ext/$krnl\_hls"
+            foreach src_dir $src_dirs {
+                if {[file isdirectory "${CMAKE_SOURCE_DIR}/$src_dir/hls"]} {
+                    set krnls [glob -nocomplain -tails -directory "${CMAKE_SOURCE_DIR}/$src_dir/hls" -type d *]
+
+                    foreach krnl $krnls {
+                        # Copy all files
+                        file mkdir "$build_dir/$project\_config_$i/user\_c$i\_$j/hdl/ext/$krnl\_hls"
+                        set files_hdl [glob -nocomplain "${CMAKE_SOURCE_DIR}/$src_dir/hls/$krnl/*.hpp" "${CMAKE_SOURCE_DIR}/$src_dir/hls/$krnl/*.cpp" "${CMAKE_SOURCE_DIR}/$src_dir/hls/$krnl/*.h" "${CMAKE_SOURCE_DIR}/$src_dir/hls/$krnl/*.c"]
```

Comment:
Not part of this PR, but could we change files_hdl to files_hls or some other name?